eclair-core/src/main/scala/fr/acinq/eclair/NodeParams.scala

```diff
@@ -197,7 +199,16 @@ object NodeParams {
 
     val features = Features.fromConfiguration(config)
     val featuresErr = Features.validateFeatureGraph(features)
+
     require(featuresErr.isEmpty, featuresErr.map(_.message))
+    require(pluginParams.forall(_.feature.mandatory > 128), "Plugin mandatory feature bit is too low, must be > 128")
+    require(pluginParams.forall(_.feature.mandatory % 2 == 0), "Plugin mandatory feature bit is odd, must be even")
+    require(pluginParams.flatMap(_.tags).forall(_ > 32768), "Plugin messages tags must be > 32768")
+    val pluginFeatureSet = pluginParams.map(_.feature.mandatory).toSet
+    require(Features.knownFeatures.map(_.mandatory).intersect(pluginFeatureSet).isEmpty, "Plugin feature bit overlaps with known feature bit")
+    require(pluginFeatureSet.size == pluginParams.size, "Duplicate plugin feature bits found")
+
+    val coreAndPluginFeatures = features.copy(activated = features.activated ++ pluginParams.map(_.activatedFeature))
```

Comment:
Is there a reason you can't use `unknown` set? It looks like it would be a good fit:

```suggestion
    val coreAndPluginFeatures = features.copy(unknown = pluginParams.map(p => UnknownFeature(p.feature.optional)).toSet)
```